docs/framework/templates/index.md

```diff
@@ -0,0 +1,278 @@
+
+Keep in mind that when you modify a template, you might have a template
+that doesn't work properly anymore after an upgrade of CiviCRM, because
+the layout has changed or the name of variables assigned to the template
+was modified. In our experience, the easiest is to use a source code
+management system (SCM) to keep track  of the changes you have made.
+Before doing any modification of the template you copied, add it to your
+SCM, and obviously also commit the template after having modified it.
+That way, you can easily generate a patch of your changes, and see how
+to apply them to the latest version of the template.
+
+## Semantically meaningful HTML attributes
+
+To make it as easy as possible for you to style any element in the page
+(e.g. put a yellow background on all the contacts of the subtype
+"members"), or add Ajax (clicking on the status of the activity changes
+it to complete), we strive to have a consistent and coherent schema for
+class names and ids for the generated HTML. This makes it easier to
+isolate the elements you want to alter from a custom style or from
+JavaScript:
+
+-   There is a class `crm-entityName` defining the type of the entity
+    bubbled up as high as possible in the DOM. For instance, each line
+    on a list of activity has `<tr class="crm-activity ...">`
+-   There is an id `crm-entityName_entityID` allowing to find the id
+    of the entity bubbled up. e.g., on a list of contacts, the
+    contact number 42  has a `<tr id="crm-contact_42" ...>`
+-   Each field or column contains a class identifying it, e.g.,
+    `"crm-activity-subject"`
+-   Each field or column that contains a value with a fixed set of
+    possible values (e.g., a Status, a Role, a Contact Type) contains a
+    class identifying it. It doesn't contain the human readable version
+    (that can be changed), but the id or a name that can't be modified
+    by the end-user; such as `class="crm-activity-status-id_42"`.
+    This is on the top of the class identifying the field name, so the
+    complete HTML is `<td class="crm-activity-status
+    crm-activity-status-id_42">Hitchhiked</td>`.
+
+At the time of the writing, some of the templates don't follow these
+conventions. Please update them and submit a bug tracking issue with a
+patch if you need to use a template that isn't yet complying. For more
+information about submitting a bug or issue, read about the CiviCRM
+community.
+
+## Displaying more content and adding Ajax features
+
+If your modifications go further than "simple" modifications of the
+layout, but need to display more content than the one assigned to the
+template by default, or to add Ajax functionality, use the CiviCRM API.
+Please read more information about using the CiviCRM API from Ajax to
+pursue this approach.
+
+In most cases, using the CiviCRM APIs should be simple and only takes a
+few extra lines of modifications.
+
+## Appending jQuery or other code to a template
```

Comment:
Better to use hooks and `CRM_Core_Resources::singleton()->addScriptFile` here.  Do we have a part of the docs that explains this?